tasks/release.py

```diff
@@ -78,13 +84,163 @@
+) -> dict:
+    """Change the context to the agent6 branch.
+
+    The branch can be overriden by setting the AGENT6_BRANCH environment variable.
+
+    Note:
+        Should be used only for operations modifying agent6 files.
+        Prefer using agent_context for read only operations.
+
+    Returns:
+        Dict of {'branch_switched', 'stashed', 'base_branch', 'branch'}
+    """
+
+    global is_agent6_context
+
+    base_branch = get_current_branch(ctx)
+
+    if version is None and major_version is None:
+        version = DEFAULT_AGENT6_VERSION
+        branch = DEFAULT_AGENT6_BRANCH
+    elif major_version is not None:
+        version = version or DEFAULT_AGENT6_VERSION
+        branch = DEFAULT_AGENT6_BRANCH
+    else:
+        branch = version[:4] + '.x'
+
+    branch = os.getenv('AGENT6_BRANCH', branch)
+
+    check_version(version, allow_agent6=True)
+    assert version.startswith('6.'), 'Not an agent6 version'
+    # Ensure this branch exists
+    assert len(ctx.run(f'git branch --list {branch}', hide=True).stdout.strip()), f"Branch {branch} does not exist"
+
+    info = {'branch_switched': False, 'stashed': False, 'base_branch': base_branch, 'branch': branch}
+
+    # Already on the target branch
+    if base_branch == branch:
+        return info
+
+    should_stash = check_uncommitted_changes(ctx)
+
+    assert not (should_stash and not allow_stash), "Uncommitted changes detected, aborting"
+
+    # Save + change branch
+    if should_stash:
+        print(color_message("Stashing uncommitted changes", "bold"))
+        print(f'{color_message("Agent6", "bold")}: Stashing uncommitted changes')
+        ctx.run("git stash -u", hide=True)
+        info['stashed'] = True
+    print(f'{color_message("Agent6", "bold")}: Checking out to {branch}')
+    ctx.run(f"git checkout {branch}", hide=True)
+    info['branch_switched'] = True
+    is_agent6_context = True
+
+    if echo_switch_info:
+        print(f'{color_message("Agent6", "bold")}: Switched to {branch} branch to perform agent6 operations')
+
+    return info
+
+
+@contextmanager
+def agent_context(ctx, version: str | None = None, major_version: int | None = None, mutable=False):
+    """If the version is 6.XX.X, checkout temporarily (or constantly if `mutable`) to the 6.XX.x branch (or $AGENT_6_BRANCH if set).
+
+    Args:
+        version: Agent full version ('6.53.0', '7.42.2-rc.1', etc.).
+        mutable: Won't leave the context and will disallow squashes if set to True.
+
+    Example:
+        > with agent_context(ctx, "6.53.0"):
+        >     ctx.run("git status")  # 6.53.x branch
     """
 
+    global is_agent6_context
+
+    if not major_version and not version:
+        major_version = 6
+
+    if not major_version:
+        version = version or DEFAULT_AGENT6_VERSION
+
+        check_version(version, allow_agent6=True)
+
+    if major_version == 6 or (version and version.startswith('6.')):
+        was_agent6_context = is_agent6_context
```

Comment:
All this logic is complex to understand, it would deserve some tests to clarify.
Besides, regarding what you are doing you could set default values for `major_version` to `6` and `version` to `DEFAULT_AGENT6_VERSION`, wdyt?